spyder_line_profiler/spyder/widgets.py

```diff
@@ -100,123 +100,214 @@ def natural_sort_key(key):
+            tip=_('Save line profiling data'),
+            icon=self.create_icon('filesave'),
+            triggered=self.save_data,
+        )
+        self.clear_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Clear,
+            text=_("Clear output"),
+            tip=_('Clear'),
+            icon=self.create_icon('editdelete'),
+            triggered=self.clear_data,
+        )
 
-        layout = QVBoxLayout()
-        layout.addLayout(hlayout1)
-        layout.addLayout(hlayout2)
-        layout.addWidget(self.datatree)
-        self.setLayout(layout)
-
-        self.process = None
         self.set_running_state(False)
-        self.start_button.setEnabled(False)
-
+        self.start_action.setEnabled(False)
+        self.clear_action.setEnabled(False)
+        self.log_action.setEnabled(False)
+        self.save_action.setEnabled(False)
+        
+        # Main Toolbar
+        toolbar = self.get_main_toolbar()
+        for item in [self.filecombo, self.browse_action, self.start_action, 
+                     self.stop_action]:
+            self.add_item_to_toolbar(
+                item,
+                toolbar=toolbar,
+                section=SpyderLineProfilerWidgetMainToolbarSections.Main,
+            )
+
+        # Secondary Toolbar
+        secondary_toolbar = self.create_toolbar(
+            SpyderLineProfilerWidgetToolbars.Information)
+        for item in [self.collapse_action, self.expand_action,
+                     self.create_stretcher(
+                         id_=SpyderLineProfilerWidgetInformationToolbarItems.Stretcher1),
+                     self.datelabel,
+                     self.create_stretcher(
+                         id_=SpyderLineProfilerWidgetInformationToolbarItems.Stretcher2),
+                     self.log_action,
+                     self.save_action,
+                     self.clear_action]:
+            self.add_item_to_toolbar(
+                item,
+                toolbar=secondary_toolbar,
+                section=SpyderLineProfilerWidgetInformationToolbarSections.Main,
+            )
+        
         if not is_lineprofiler_installed():
-            for widget in (self.datatree, self.filecombo, self.log_button,
-                           self.start_button, self.stop_button, browse_button,
-                           self.collapse_button, self.expand_button):
+            for widget in (self.datatree, self.filecombo, self.log_action,
+                           self.start_action, self.stop_action, self.browse_action,
+                           self.collapse_action, self.expand_action):
                 widget.setDisabled(True)
             text = _(
                 '<b>Please install the <a href="%s">line_profiler module</a></b>'
                 ) % WEBSITE_URL
             self.datelabel.setText(text)
             self.datelabel.setOpenExternalLinks(True)
         else:
-            pass  # self.show_data()
+            pass  
```

Comment:
Remove trailing whitespace:
```suggestion
            pass
```